test/common/http/client_test.cc

```diff
@@ -725,11 +800,15 @@ TEST_F(ClientTest, EncodeMetadata) {
   createStream();
 
   // Send request headers.
+  EXPECT_CALL(dispatcher_, pushTrackedObject(_));
+  EXPECT_CALL(dispatcher_, popTrackedObject(_));
   envoy_headers c_headers = defaultRequestHeaders();
   EXPECT_CALL(request_decoder_, decodeHeaders_(_, true));
   http_client_.sendHeaders(stream_, c_headers, true);
 
   // Encode response headers.
+  EXPECT_CALL(dispatcher_, pushTrackedObject(_));
+  EXPECT_CALL(dispatcher_, popTrackedObject(_));
```

Comment:
Don't need to block on this, but we might just want to adopt `NiceMock` behavior for these calls so we don't have to count them everywhere in tests.